zfit/models/dist_tfp.py

```diff
@@ -714,3 +718,120 @@ class StudentTPDFRepr(BasePDFRepr):
+        It is defined for q < 3 and the Gaussian distribution is recovered as q -> 1.
+        For q < 1, is it the PDF of a bounded random variable.
+        We only support 1 < q < 3 in this implementation.
+        If you want to use exactly q = 1, use the `zfit.pdf.Gauss` class.
+        During fitting, if you want to start from a Gaussian shape, you can initialize the `q` parameter to be really close to 1.
+        It is related to the Student's t-distribution according to the `corresponding Wikipedia entry <https://en.wikipedia.org/wiki/Q-Gaussian_distribution#Student's_t-distribution>`_
+        and that is how it is implemented here.
+
+        The q-Gaussian shape for 1 < q < 3 is defined as
+
+        .. math::
+
+            f(x \\mid q, \\mu, \\sigma) = \\frac{1}{C_{q} \\sigma} e_{q}\\left(-\\left(\\frac{x - \\mu}{\\sigma}\\right)^{2}\\right)
+
+        with
+
+        .. math::
+
+            e_q(x) = \\left[1 + (1 - q) x\\right]_{+}^{\\frac{1}{1 - q}}
+
+        and the normalization over [-inf, inf] of
+
+        .. math::
+
+            C_{q} = \\frac{\\sqrt{\\pi} \\Gamma \\left(\\frac{3 - q}{2 (q - 1)}\\right)}{\\sqrt{q - 1}\\Gamma \\left(\\frac{1}{q - 1}\\right)}
+
+        The normalization changes for different normalization ranges
+
+        Args:
+            q: Shape parameter of the q-Gaussian. Must be 1 < q < 3.
+            mu: Mean of the distribution
+            sigma: Scale of the distribution
+            obs: |@doc:model.init.obs| Observables of the
+               model. This will be used as the default space of the PDF and,
+               if not given explicitly, as the normalization range.
+
+               The default space is used for example in the sample method: if no
+               sampling limits are given, the default space is used.
+
+               The observables are not equal to the domain as it does not restrict or
+               truncate the model outside this range. |@docend:model.init.obs|
+            extended: |@doc:pdf.init.extended| The overall yield of the PDF.
+               If this is parameter-like, it will be used as the yield,
+               the expected number of events, and the PDF will be extended.
+               An extended PDF has additional functionality, such as the
+               ``ext_*`` methods and the ``counts`` (for binned PDFs). |@docend:pdf.init.extended|
+            norm: |@doc:pdf.init.norm| Normalization of the PDF.
+               By default, this is the same as the default space of the PDF. |@docend:pdf.init.norm|
+            name: |@doc:model.init.name| Human-readable name
+               or label of
+               the PDF for better identification.
+               Has no programmatical functional purpose as identification. |@docend:model.init.name|
+        """
+        q, mu, sigma = self._check_input_params(q, mu, sigma)
+        if q < 1 or q > 3:
```

Comment:
Sorry, somehow my comment previously here disappeared (maybe I forgot to send): we should also test with tf.debugging.assert, at least once it's not executing eagerly.
```
if run.executing_eagerly():
    if q < 1 or q > 3:...
    # the rest
elif run.numeric_checks:
    tf.debugging.assert...
    ```
The reason is that if this is created _inside_ a TF function, it will otherwise fail. I think python checks are in general better, the error is nicer etc, but if it happens inside a tf.function, the check will fail (i.e. it will error, because `q` is a symbolic tensor and doesn't have a truth value while Python runs)